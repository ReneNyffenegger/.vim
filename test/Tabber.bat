@del Tabber.out
@gvim -c "source Tabber.vim"
@fc Tabber.out Tabber.expected
