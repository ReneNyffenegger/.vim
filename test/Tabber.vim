"
"    Use Tabber.bat to Â«callÂ» this file.
"

"   Change to = 1 to test manually, without Tabber.bat
let s:manually = 0

fu! <SID>InsertFooBarBaz() " {

"    \ ['ins-const', 'Foo. press tab again '],
   call Tabber#Add([
     \ ['ins-const', 'Bar. type 123 '       ],
     \ ['ins-const', 'Baz. press tab again ']
     \ ])

   return 'Foo. press tab again '

endfu " }

fu! <SID>InsertOneTwoThree() " {

   call Tabber#Add([
     \ ['ins-const', 'One. press tab again '  ],
     \ ['ins-const', 'Two. press tab again '  ],
     \ ['ins-const', 'Three. press tab again '],
     \ ])

"  echo "go to insertmode and press tab"

   return 'EggsWhyZee '

endfu " }

new


if mapcheck('123') == ""
  inoremap <expr> 123 <SID>InsertOneTwoThree()
endif

if mapcheck('fbb') == ""
  inoremap <expr> fbb <SID>InsertFooBarBaz()
endif

if s:manually == 1
   echo "Go to insertmode and enter fbb"
else
   execute "normal ifbb a \t b 123 \t c \t d \t e \t f \t g<"

  write! Tabber.out

  quitall!

endif
