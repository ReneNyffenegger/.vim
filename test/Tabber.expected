line 1
line 2
  IUS ENTER-1
  ius ENTER-2<
line 3
line 4
* line 5
  <h1>__H1H1__One -Two -Three </h1><!--{-->
    <div>__DIV_DIV__</div>
    after H1      <
  <!--}-->
line 6
line 7

  <h1>XXX</h1><!--{-->
    <div>YYYY</div>
    ZZZZ
  <!--}-->

line 8
     Space intentionally left blank
  EnEnEnTop [first]
  EnEnEnBottom [second third]
     One ,Two ,Three ,  <
